presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchQueryBuilder.java

```diff
@@ -0,0 +1,228 @@
+import static com.facebook.presto.spi.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.spi.type.IntegerType.INTEGER;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class ElasticsearchQueryBuilder
+{
+    private static final Logger LOG = Logger.get(ElasticsearchQueryBuilder.class);
+
+    private final Duration scrollTimeout;
+    private final int scrollSize;
+    private final TransportClient client;
+    private final int shard;
+    private final TupleDomain<ColumnHandle> tupleDomain;
+    private final List<ElasticsearchColumnHandle> columns;
+    private final String index;
+    private final String type;
+
+    public ElasticsearchQueryBuilder(List<ElasticsearchColumnHandle> columnHandles, ElasticsearchConnectorConfig config, ElasticsearchSplit split)
+    {
+        columns = requireNonNull(columnHandles, "columnHandles is null");
+        requireNonNull(config, "config is null");
+        requireNonNull(split, "split is null");
+        ElasticsearchTableDescription table = split.getTable();
+        tupleDomain = split.getTupleDomain();
+        index = split.getIndex();
+        type = table.getType();
+        shard = split.getShard();
+        try {
+            Settings settings = Settings.builder()
+                    .put("client.transport.ignore_cluster_name", true)
+                    .build();
+            client = new PreBuiltTransportClient(settings)
+                    .addTransportAddress(new TransportAddress(InetAddress.getByName(split.getSearchNode()), split.getPort()));
+        }
+        catch (IOException e) {
+            throw new PrestoException(ELASTIC_SEARCH_CONNECTION_ERROR, format("Error connecting to Elasticsearch SearchNode: %s with port: %d", split.getSearchNode(), split.getPort()), e);
+        }
+        scrollTimeout = config.getScrollTimeout();
+        scrollSize = config.getScrollSize();
+    }
+
+    public void close()
+    {
+        client.close();
+    }
+
+    public SearchRequestBuilder buildScrollSearchRequest()
+    {
+        String indices = index != null && !index.isEmpty() ? index : "_all";
+        SearchRequestBuilder searchRequestBuilder = client.prepareSearch(indices)
+                .setTypes(type)
+                .setSearchType(SearchType.QUERY_THEN_FETCH)
+                .setScroll(new TimeValue(scrollTimeout.toMillis()))
+                .setQuery(buildSearchQuery())
+                .setPreference("_shards:" + shard)
+                .setSize(scrollSize);
+        LOG.debug("ElasticSearch Request: " + searchRequestBuilder.toString());
```

Comment:
not sure how big `searchRequestBuilder` can get, but you may want to guard this log statement with `if (LOG.isDebugEnabled())`

Reply:
please do not concatenate, instead do:
```
LOG.debug("ElasticSearch Request: %s", searchRequestBuilder);
```